modules/rocksdb/src/xtdb/rocksdb.clj

```diff
@@ -13,65 +13,98 @@
            (java.nio.file Files Path)
            java.nio.file.attribute.FileAttribute
            (org.rocksdb BlockBasedTableConfig Checkpoint CompressionType FlushOptions LRUCache
-                        Options ReadOptions RocksDB RocksIterator
-                        WriteBatch WriteOptions Statistics StatsLevel)))
+                        DBOptions Options ReadOptions RocksDB RocksIterator
+                        WriteBatch WriteOptions Statistics StatsLevel
+                        ColumnFamilyOptions ColumnFamilyDescriptor ColumnFamilyHandle)))
+
+(defprotocol CfId
+  (->cf-id [this]))
+
+(extend-protocol CfId
+  (class (byte-array 0))
+  (->cf-id [this]
+    (first this))
+
+  org.agrona.DirectBuffer
+  (->cf-id [this]
+    (.getByte this 0))
+
+  java.nio.ByteBuffer
+  (->cf-id [this]
+    (.get this 0)))
 
 (set! *unchecked-math* :warn-on-boxed)
 
+(def ^:const column-family-defs [c/entity+vt+tt+tx-id->content-hash-index-id
+                                 c/entity+z+tx-id->content-hash-index-id])
+
 (defn- iterator->key [^RocksIterator i]
   (when (.isValid i)
     (mem/as-buffer (.key i))))
 
-(defrecord RocksKvIterator [^RocksIterator i]
+(defrecord RocksKvIterator [i is k->i]
```

Comment:
given this is low-level, I'd go for some lower-level, more performant primitives here:
* `defrecord` -> `deftype` (so you can use mutable fields)
* `is`: `(atom {})` -> `(HashMap.)`
* `i`: `(atom nil)` -> `^:unsynchronised-mutable ^RocksIterator i`